packages/@aws-cdk/assert/lib/assertions/have-resource-matchers.ts

```diff
@@ -318,6 +318,7 @@ export class Capture<T=any> {
   /**
    * A Capture object that captures a custom type
    */
+  // eslint-disable-next-line @typescript-eslint/no-shadow
```

Comment:
is this needed? can't quite tell why if so.

Reply:
yeah it is needed because the new typescript-eslint plugin also detects shadowing for generic names. `T` is apparently shadowed here.